src/WebJobs.Script.Grpc/Server/DefaultHttpProxyService.cs

```diff
@@ -97,7 +97,12 @@ public void StartForwarding(ScriptInvocationContext context, Uri httpUri)
             httpContext.Items[ScriptConstants.HttpProxyingEnabled] = bool.TrueString;
 
             // add invocation id as correlation id
-            httpRequest.Headers.TryAdd(ScriptConstants.HttpProxyCorrelationHeader, context.ExecutionContext.InvocationId.ToString());
+            if (!httpRequest.Headers.TryAdd(ScriptConstants.HttpProxyCorrelationHeader, context.ExecutionContext.InvocationId.ToString()))
+            {
+                // try to remove the header if it already exists
+                httpRequest.Headers.Remove(ScriptConstants.HttpProxyCorrelationHeader);
+                httpRequest.Headers.TryAdd(ScriptConstants.HttpProxyCorrelationHeader, context.ExecutionContext.InvocationId.ToString());
```

Comment:
Rather than do this, can we do `httpRequest.Headers[ScriptConstants.HttpProxyingEnabled] = ...` since we're going to overwrite it anyway? That'd be less operations.